modules/caddyhttp/server.go

```diff
@@ -911,6 +912,24 @@ type lengthReader struct {
 	Length int
 }
 
+var lengthReaderPool = sync.Pool{
+	New: func() interface{} {
+		return &lengthReader{}
+	},
+}
+
+func getLengthReader(source io.ReadCloser) *lengthReader {
+	reader := lengthReaderPool.Get().(*lengthReader)
+	reader.Source = source
+	reader.Length = 0
+	return reader
+}
+
+func putLengthReader(reader *lengthReader) {
+	reader.Source = nil
+	lengthReaderPool.Put(reader)
+}
```

Comment:
I think the length should be reset when put back in the pool rather than when getting it. Numerical types already default to zero in Go. This makes it more symmetrical, i.e. `put` doing all the cleanup.

```suggestion
func getLengthReader(source io.ReadCloser) *lengthReader {
	reader := lengthReaderPool.Get().(*lengthReader)
	reader.Source = source
	return reader
}

func putLengthReader(reader *lengthReader) {
	reader.Source = nil
	reader.Length = 0
	lengthReaderPool.Put(reader)
}
```